From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: VerteDinde <vertedinde@electronjs.org>
Date: Thu, 3 Mar 2022 23:06:06 -0800
Subject: build: disable -no-standalone-debug

After upgrading to Xcode 13.2.1, generating dSYMs for breakpad began failing on:
'the segment '__DWARF' claims its contents lie beyond the end of the file'.
This error is similar to the original CR bug for dsymutil and a 4GB+ dsym file
that required the -no-standalone-debug flag. This patch removes the flag to fix
this error.

I actually think removing this flag will increase the dSYM file size and make
the issue worse, but lets try it and see.

diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index 7b9ce0172eedf2b55fe7f3de7baee2f2028f4a7b..8d8ef5930dbd0659a6c0951fa446705b13d2a94a 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2312,14 +2312,14 @@ config("symbols") {
     configs = [ ":win_pdbaltpath" ]
   } else {
     cflags = []
-    if (is_mac && enable_dsyms) {
+    # if (is_mac && enable_dsyms) {
       # If generating dSYMs, specify -fno-standalone-debug. This was
       # originally specified for https://crbug.com/479841 because dsymutil
       # could not handle a 4GB dSYM file. But dsymutil from Xcodes prior to
       # version 7 also produces debug data that is incompatible with Breakpad
       # dump_syms, so this is still required (https://crbug.com/622406).
-      cflags += [ "-fno-standalone-debug" ]
-    } else if (is_mac && !use_dwarf5) {
+      # cflags += [ "-fno-standalone-debug" ]
+    if (is_mac && !use_dwarf5) {
       # clang defaults to DWARF2 on macOS unless mac_deployment_target is
       # at least 10.11.
       # TODO(thakis): Remove this once mac_deployment_target is 10.11.
